### ISSUE TITLE
Can we simplify the build output?

### ISSUE THREAD

**Issue by willstott101** [dojo/cli-build-app#267]
**Enhancement**

Package Version: 5.0.2

I'm trying to integrate dojo into a simple Django app. Ideally I would like to be able to simply ask dojo to dump it's built JS and CSS files into a static files folder without any hashes in the file name.

I don't like the idea of dropping the dojo cli tooling completely to manage this, and I don't like the idea of working around the unpredictable file names in other specific build steps.

I have two questions:

* Would you be interested in a PR adding a config/cmdline option for not including a hash in the file names (as is done currently for dev builds)?
  * e.g. `--no-hash`
  * I would like to manage cache control in the same way as the rest of the project.
* Would you be interested in a PR adding a config/cmdline option for minimising build output (no source maps, no instrumentation, just the run-time required files)?
  * e.g. `--minimal`

**Comment by matt-gadd** [comment]
hi @willstott101, thanks for taking the time to raise the issue 👍. It makes sense that a unique file hash might not be desirable in certain situations (especially for other tooling).

I think out of the 2 options listed a good and non controversial start would be option a) `--no-hash` as I think it's clear what that turns off, while `--minimal` is a bit more cryptic and would probably require a bit more thought on what we class as "minimal". 

We'd definitely love a PR to add this! 🎉

**Comment by willstott101** [comment]
Great thanks, I'm taking a look now.

Thanks to Yargs, `--no-hash` [actually turns out to be a little complicated](https://github.com/yargs/yargs/issues/1307).

That's ok if I do something like this:
![Screenshot from 2019-04-30 13-00-22](https://user-images.githubusercontent.com/335152/56960208-f643cf80-6b47-11e9-9eca-821fdb9597e5.png) but I can't see any similar pattern used in dojo CLIs

Also I suppose for this option to make sense the dev target should honor `--hash` but keep it's default to not hash? That makes the help text from yargs even more confusing.

**Comment by matt-gadd** [comment]
@willstott101 ha i didn't realise yargs had actual negation prefix functionality built in. and yeah having the `--hash` flag is going to be a bit confusing for other modes like you said (I don't think we have any plans to offer a hash in other modes). perhaps we can just cheat with something like:
```
--exclude-hash     Does not use file hashes for output in dist mode
```
or something like that?

**Comment by willstott101** [comment]
Yep that seems reasonable to me. The Yarg's negation stuff can be configured but then that's a separate patch to `@dojo/cli` I think. I'll go with `--omit-hash` in my PR I think, _exclude_ feels like we're ignoring input to me.

**Comment by matt-gadd** [comment]
@willstott101 `--omit-hash` sound perfect to me 👍

**Comment by matt-gadd** [comment]
@willstott101 sorry for the slight delay, but now published to npm at `5.0.4`. Thanks again 😄

**Comment by willstott101** [comment]
No worries, thanks for the speedy responses!
